packages/web/src/builders/file-server/file-server.impl.ts

```diff
@@ -1,9 +1,10 @@
 import { exec, execSync } from 'child_process';
-import { ExecutorContext } from '@nrwl/devkit';
+import { ExecutorContext, joinPathFragments } from '@nrwl/devkit';
 import ignore from 'ignore';
 import { readFileSync } from 'fs';
 import { Schema } from './schema';
 import { watch } from 'chokidar';
+import { workspaceLayout } from '@nrwl/workspace/src/core/file-utils';
```

Comment:
@vsavkin is this the best way to get the `workspaceLayout`? `@nrwl/devkit` only exports similar functions that require a `Tree` (e.g. during generator runs). And some duplicate it into their package (which is also an option in case)

Reply:
Sorry for interfering here. I think it is the best option as it reads the content of the `nx.json` file and falls back to `apps` and `libs` when not present. 

```ts
export function getWorkspaceLayout(host: Tree): { appsDir: string; libsDir: string; npmScope: string } {
  const nxJson = readJson<NxJsonConfiguration>(host, 'nx.json');
  return {
    appsDir: nxJson.workspaceLayout?.appsDir ?? 'apps',
    libsDir: nxJson.workspaceLayout?.libsDir ?? 'libs',
    npmScope: nxJson.npmScope,
  };
}
```

Reply:
@juristr Sorry i misread the import. I think you should use `getWorkspaceLayout` from `@nrwl/devkit`

Reply:
The one from `@nrwl/workspace` is the best to use here. :+1: